atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/sweep/queue/TargetedSweeperTest.java

```diff
@@ -964,25 +968,59 @@ public void doesNotLeaveSentinelsIfTableDestroyed() {
         unreadableTs = 11;
         spiedKvs.truncateTable(TABLE_CONS);
         assertThat(spiedKvs.getRange(TABLE_CONS, RangeRequest.all(), Long.MAX_VALUE)).isEmpty();
-        sweepQueue.sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
+        sweepNextBatch(ShardAndStrategy.conservative(CONS_SHARD));
         assertThat(spiedKvs.getRange(TABLE_CONS, RangeRequest.all(), Long.MAX_VALUE)).isEmpty();
     }
 
+    @Test
+    public void sweepOnlyOneFinePartitionByDefault() {
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS);
+        enqueueTombstone(TABLE_CONS, LOW_TS + 2);
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS + 4);
+        enqueueTombstone(TABLE_CONS, LOW_TS + 6);
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS + 8);
+        // write in the next fine partition
+        enqueueWriteCommitted(TABLE_CONS, maxTsForFinePartition(0) + 1);
+        enqueueTombstone(TABLE_CONS, maxTsForFinePartition(0) + 2);
+
+        sweepQueue.processShard(ShardAndStrategy.conservative(CONS_SHARD));
+
+        assertTestValueEnqueuedAtGivenTimestampStillPresent(TABLE_CONS, maxTsForFinePartition(0) + 1);
+    }
+
+    @Test
+    public void batchShardIterationsSweepsMultipleFinePartitions() {
+        batchShardIterations = true;
+
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS);
+        enqueueTombstone(TABLE_CONS, LOW_TS + 2);
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS + 4);
+        enqueueTombstone(TABLE_CONS, LOW_TS + 6);
+        enqueueWriteCommitted(TABLE_CONS, LOW_TS + 8);
+        // write in the next fine partition
+        enqueueWriteCommitted(TABLE_CONS, maxTsForFinePartition(0) + 1);
+        enqueueTombstone(TABLE_CONS, maxTsForFinePartition(0) + 2);
+
+        sweepQueue.processShard(ShardAndStrategy.conservative(CONS_SHARD));
+
+        assertReadAtTimestampReturnsSentinel(TABLE_CONS, maxTsForFinePartition(0) + 1);
```

Comment:
Non-actionable: Would be nice to verify that the timestamp suppliers are only invoked once even though multiple batches got processed, but unfortunately this looks like it might need some refactoring of the abstract test